### PR TITLE
mgr/dashboard: fix API docs link

### PR DESCRIPTION
**Pull request by avanthakkar**
Fixes the broken API docs link after the updated changes introduced [here](https://github.com/ceph/ceph/pull/41345)

Fixes: https://tracker.ceph.com/issues/50890
Signed-off-by: Avan Thakkar <athakkar@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by avanthakkar** [comment]
This PR just intends to fix the broken API docs link. @aaSharma14 has already started the work for creating the separate frontend component as discussed. Can you confirm @aaSharma14 ?

**Comment by aaSharma14** [comment]
> This PR just intends to fix the broken API docs link. @aaSharma14 has already started the work for creating the separate frontend component as discussed. Can you confirm @aaSharma14 ?

Yes, I have started working on the separate component.

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/dashboard-help/dashboard-help.component.html [hunk 1]
```diff
@@ -1,9 +1,7 @@
 <form #docsForm
       action="/docs"
       target="_blank"
-      method="post">
-  <input type="hidden"
-         name="token" />
+      method="get">
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/dashboard-help/dashboard-help.component.html:4]: Can we get rid of the form (and also ```goToApiDocs()```) and replace it with a link?

Comment (avanthakkar) [reply src/pybind/mgr/dashboard/frontend/src/app/core/navigation/dashboard-help/dashboard-help.component.html]: Done!

**Comment by avanthakkar** [comment]
jenkins test dashboard

**Review by nizamial09 — APPROVED** [review]

**Review by aaSharma14 — APPROVED** [review]

**Review by pereman2 — APPROVED** [review]

**Review by epuertat — APPROVED** [review]

**Pull request merged**